yourbench/pipeline/handler.py

```diff
@@ -127,10 +132,11 @@ def run_pipeline(
         stage_start_time: float = time.time()
 
         # Ensure the specific stage config is at least an empty dict if it was None
-        if stage_name in config.get("pipeline", {}) and config["pipeline"][stage_name] is None:
-            config["pipeline"][stage_name] = {}
+        if pipeline_config[stage_name] is None:
```

Comment:
This will fail if you are not using a default dict, no?
Maybe use get or catch the KeyError